### PR TITLE
fix: attempt to fix building for Android

### PR DESCRIPTION
**Pull request by ErikBjare**
fixes https://github.com/ActivityWatch/aw-server-rust/issues/211

### CODE REVIEW

File: aw-server/Cargo.toml [hunk 1]
```diff
@@ -35,10 +35,10 @@ aw-models = { path = "../aw-models" }
 aw-transform = { path = "../aw-transform" }
 aw-query = { path = "../aw-query" }
 
-[target.'cfg(target_os="linux")'.dependencies]
+[target.'cfg(all(target_os="linux", target_arch="x86"))'.dependencies]
```
Comment (ErikBjare) [inline aw-server/Cargo.toml:38]: @johan-bjareholt I assume this is what you intended?

File: compile-android.sh [hunk 2]
```diff
@@ -33,5 +33,6 @@ for archtargetstr in \
         cargo build -p aw-server --target $target --lib $($RELEASE && echo '--release')
     else
         echo "Couldn't find directory for target $arch"
+        exit 1
```
Comment (ErikBjare) [inline compile-android.sh:36]: Prevents silently failing builds

File: install-ndk.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@ project_path="$(readlink -f "$script_dir/.")"
 
 if [ -z "$ANDROID_NDK_HOME" ]; then
     echo 'ANDROID_NDK_HOME not set, downloading NDK...';
-    wget --no-verbose -O android-ndk.zip https://dl.google.com/android/repository/android-ndk-r20-linux-x86_64.zip;
+    wget --no-verbose -O android-ndk.zip https://dl.google.com/android/repository/android-ndk-r21-linux-x86_64.zip;
```
Comment (ErikBjare) [inline install-ndk.sh:12]: Hopefully fixes crashes on Android 11

File: install-ndk.sh [hunk 4]
```diff
@@ -40,18 +40,18 @@ echo "Creating cargo config..."
 mkdir -p $project_path/.cargo
 echo "
 [target.aarch64-linux-android]
-ar = '$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android26-ar'
+ar = '$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-ar'
```
Comment (ErikBjare) [inline install-ndk.sh:43]: Not sure if this binary rename happened with the new NDK, but I assume so.

File: aw-server/Cargo.toml (hunk 1 above)
Comment (johan-bjareholt) [reply aw-server/Cargo.toml:38]: Correct!

**Pull request merged**